### PR TITLE
Add ability to populate env from secret & various bugfixes

### LINKED ISSUE #193: Extra environment variables from secrets
## Use Case
Currently we can mount extra environment variables to all of the containers that the chart brings up. However, there is no method add environment variables from secrets which could be used in things such as custom entry points.

## Describe the Solution You Would Like
Additional value, e.g.:

```
puppetserver:
  masters:
    extraEnvSecret: "secretName"
```

This would add the following to the `puppetserver` container:
```
{{- if .Values.puppetserver.masters.extraEnvSecret }}
envFrom:
  - secretRef:
    name: {{ .Values.puppetserver.masters.extraEnvSecret }}
{{- end }}
```

## Describe Alternatives You've Considered
I've tried adding secrets via `extraSecrets` but these aren't mounted to the containers and would still require to parse them in bash (or similar) to retrieve values for specific secrets that I need in my custom entrypoint scripts. Getting them as environment variables would be the proper way.

## Additional Context
I can create a PR for this.

### PR DESCRIPTION
**Pull request by ldaneliukas**
# Feature
Fixes #193 

Tested by deploying our usual setup from this branch - helm reports no changes to any resources. `envFrom` is not populated:
```
 ~  k get pod puppetserver-puppetserver-master-7458c5f9f7-qcms6 -oyaml | grep envFrom -A2
```

Redeployed after adding the following to Helm values:
```
puppetserver:
  masters:
    extraEnvSecret: puppetserver-env
```

Which resulted in the following changes:
```
 ~  k get pod puppetserver-puppetserver-master-85946b589c-rrsnl -oyaml | grep envFrom -A2
    envFrom:
    - secretRef:
        name: puppetserver-env
--
    envFrom:
    - secretRef:
        name: puppetserver-env
```

By executing `k exec -ti puppetserver-puppetserver-master-85946b589c-rrsnl /usr/bin/env` I am able to see all of the environment variables populated from secret `puppetserver-env` key/value pairs.

## Usage example

One of the possible uses is to create custom entrypoints with secret values. For example, some internal Puppet function might need a file with credentials:
```
puppetserver:
  customentrypoints:
    enabled: true
    configmaps:
      01_some_dependencies.sh: |-
        #!/bin/bash
        echo "{\"client_id\":\"${JWT_CLIENT}\",\"client_secret\":\"${JWT_SECRET}\"}" > /etc/jwt_credentials.json
```

I'm sure there's many other use cases for this.

# Bugfixes
- There is no readinessProbePeScheme, probably leftover when copy+pasting. Fixing to readinessProbeScheme.
- Fix PuppetDB [defaulting](https://github.com/voxpupuli/container-puppetdb/blob/main/puppetdb/Dockerfile#L33) to deprecated `-Xloggc` flag and using a non existent path for GC logs.
- Broken r10k-code command for statefulset compilers breaks chart when using them
- Standardize r10k-code readiness probe usage as it is different across deployment methods

### CODE REVIEW

**Comment by Xtigyro** [comment]
@ldaneliukas Some conflicts need resolving before we can merge this.

Also - would you like to be added as one of the official maintainers of the chart?

CC: @davidphay

**Comment by ldaneliukas** [comment]
> @ldaneliukas Some conflicts need resolving before we can merge this.
> 
> Also - would you like to be added as one of the official maintainers of the chart?
> 
> CC: @davidphay

Solved the conflicts and merged the changes from https://github.com/puppetlabs/puppetserver-helm-chart/pull/198 into this PR to prevent more version conflicts.

Yes, I'd be happy to.

**Comment by Xtigyro** [comment]
@ldaneliukas Great!
Would you mind please adding yourself to:
1. https://github.com/puppetlabs/puppetserver-helm-chart/blob/master/Chart.yaml#L18
2. https://github.com/puppetlabs/puppetserver-helm-chart/blob/master/README.md#credits
3. https://github.com/puppetlabs/puppetserver-helm-chart/blob/master/CODEOWNERS#L3

@underscorgan Could you please give @ldaneliukas `Write` permissions and `Collaborator` status in this repo?

**Comment by ldaneliukas** [comment]
> @ldaneliukas Great! Would you mind please adding yourself to:
> 
> 1. https://github.com/puppetlabs/puppetserver-helm-chart/blob/master/Chart.yaml#L18
> 2. https://github.com/puppetlabs/puppetserver-helm-chart/blob/master/README.md#credits
> 3. https://github.com/puppetlabs/puppetserver-helm-chart/blob/master/CODEOWNERS#L3
> 
> @underscorgan Could you please give @ldaneliukas `Write` permissions and `Collaborator` status in this repo?

Done.

**Comment by Xtigyro** [comment]
FYI: Bumped the version to `8.2.0` - just following more strictly the Semantic Versioning.

**Review by Xtigyro — APPROVED** [review]

**Comment by binford2k** [comment]
I just discovered this PR, @ldaneliukas can you fill out the form at https://www.puppet.com/ecosystem/contribute/trusted-contributors/apply so we can get you onboarded to the trusted contributor program?

**Comment by ldaneliukas** [comment]
@binford2k done 👍

**Pull request merged**
